boa/src/environment/function_environment_record.rs

```diff
@@ -60,37 +61,43 @@ pub struct FunctionEnvironmentRecord {
 }
 
 impl FunctionEnvironmentRecord {
-    pub fn bind_this_value(&mut self, value: Value) {
+    pub fn bind_this_value(&mut self, value: Value) -> Result<Value, ErrorKind> {
         match self.this_binding_status {
             // You can not bind an arrow function, their `this` value comes from the lexical scope above
             BindingStatus::Lexical => {
-                // TODO: change this when error handling comes into play
                 panic!("Cannot bind to an arrow function!");
             }
             // You can not bind a function twice
-            BindingStatus::Initialized => {
-                // TODO: change this when error handling comes into play
-                panic!("Reference Error: Cannot bind to an initialised function!");
-            }
+            BindingStatus::Initialized => Err(ErrorKind::ReferenceError(
+                "Cannot bind to an initialised function!".into(),
+            )),
 
             BindingStatus::Uninitialized => {
-                self.this_value = value;
+                self.this_value = value.clone();
                 self.this_binding_status = BindingStatus::Initialized;
+                Ok(value)
             }
         }
     }
+
+    pub fn get_super_base(&self) -> Value {
+        let home = &self.home_object;
+        if home.is_undefined() {
+            Value::Undefined
+        } else {
+            assert!(home.is_object());
+            home.as_object().unwrap().prototype_instance().clone()
```

Comment:
I think there is no need to clone this, and should remove the clippy warning.
```suggestion
            home.as_object().unwrap().prototype_instance()
```

Reply:
`prototype_instance()` method returns a `&Value` and not `Value`. So we should be cloning it right?

Reply:
The `as_object()` method returns a `GcObject`, and this `GcObject` returns a `Value` (already cloned), so no need to clone. Also, I would recommend to use `expect()` instead of `unwrap()`.